omnibus/files/private-chef-cookbooks/private-chef/resources/pg_upgrade.rb

```diff
@@ -235,7 +235,7 @@ def update_to_latest_version
 
         # The new data directory may have just been created; that's why
         # this needs to be evaluated lazily
-        new_version = node['private_chef']['postgresql']['version']
+        new_version = version_from_data_dir(new_data_dir)
```

Comment:
This will not work since for Postgres 10 onwards the PG_VERSION is only single number. https://pgpedia.info/p/PG_VERSION.html

Reply:
Will wait for Shahid to report on this.

Reply:
What we did in chef-backend was a) list all directories (version dirs) in the PG root; then b) sort so that the highest (numerically) is first - that makes the first entry the new version (10.x), and the next entry the previous version (9.x). That might be a helpful approach here as well.

Reply:
@marcparadise - this is slightly different way oof doing that: https://github.com/chef/chef-server/pull/2577/commits/b553cf6caa293a837a44227dfccd4f16041fb3ff#diff-8ca7322715caf712550232f26b6ce89cf5315327452b7e79debd42b985413513R212-R214

Reply:
@marcparadise , Basically this function needs to fetch the version from the given input data directory not the latest version of the base version in the box.